integration/test/cwl_util.go

```diff
@@ -75,6 +74,62 @@ func ValidateLogs(t *testing.T, logGroup, logStream string, numExpectedLogs int,
 	assert.Equal(t, numExpectedLogs, numLogsFound)
 }
 
+// ValidateLogsInOrder takes a log group, log stream, a list of specific log lines and a timestamp.
+// It should query the given log stream for log events, and then confirm that the log lines that are
+// returned match the expected log lines.
+func ValidateLogsInOrder(t *testing.T, logGroup, logStream string, logLines []string, since time.Time) {
```

Comment:
The code here looks very similar to the original `ValidateLogs` function, and I did start to consolidate them, but decided against it because they validate different things, and breaking that out to generalize this pattern seemed like it would cause more confusion than anything else. Namely, this function validates that the log events we get from GetLogEvents matches exactly, in order, the provided `logLines` input, whereas the original validation function only checks the number of log events. Another difference, though not as big, is that this uses the `StartFromHead` API parameter to sort the events chronologically when retrieving them from CWL.

Basically, I didn't want to create a generalized pattern for doing "validation" inside a validation function, like:
```go
func ValidateLogs(t *testing.T, ..., fn func(events []OutputLogEvents) bool) {
  // get log events since timestamp
  evts := paginagateAndDoStuff()

  // perform generalized validation
  if !fn(evts) {
     t.Fail()
  }
}
```
Though cleaner than duplicating this code, I didn't think it provided much value. If anyone feels strongly that I should still consolidate, I can do that. I'm just explaining why I stopped myself from doing it before publishing the PR

Reply:
I think this is fine.